api/pkg/gsutil/gsutil.go

```diff
@@ -0,0 +1,125 @@
+package gsutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/bboughton/gcp-helpers/gsurl"
+	"github.com/pkg/errors"
+	"google.golang.org/api/option"
+)
+
+// parseURL exists to allow us to use 'url' as a request param
+var parseURL = url.Parse
```

Comment:
Because func (g *gsutil) ParseURL(url string) also uses `url` with `string` data type as input. If we try to directly use url.Parse() in ParseURL(), we will get this error: `url.Parse undefined (type string has no field or method Parse)` because `url` here is the input variable `url`, not `net/url` package.

Other options are:

1. use alias import:

```
import (
    gourl "net/url"
)

...

func (g *gsutil) ParseURL(url string) (*URL, error) {
	u, err := gourl.Parse(url)
```

Or, 2. rename the input variable:

```
func (g *gsutil) ParseURL(gsURL string) (*URL, error) {
	u, err := url.Parse(gsURL)
```

I'll go with second option.